src/app/notifications/components/SearchFailure/index.tsx

```diff
@@ -32,18 +32,20 @@ const SearchFailure = ({ dismiss, mobileToolbarOpen, selectedHighlight }: Props)
     }
   };
 
-  const resetErrorClearing = useTimeout(clearErrorAfter, startFadeOut);
+  const resetErrorClearing = useTimeout(clearErrorAfter, startFadeOut, [shouldAutoDismiss]);
   const resetAutoDismiss = useTimeout(shouldAutoDismissAfter, () => setShouldAutoDismiss(true));
 
-  useOnDOMEvent(window, !isFadingOut, 'click', startFadeOut);
-  useOnDOMEvent(window, !isFadingOut, 'scroll', startFadeOut);
+  useOnDOMEvent(window, !isFadingOut && shouldAutoDismiss, 'click', startFadeOut, [shouldAutoDismiss]);
+  useOnDOMEvent(window, !isFadingOut && shouldAutoDismiss, 'scroll', startFadeOut, [shouldAutoDismiss]);
 
   React.useEffect(() => {
-    setIsFadingOut(false);
-    setShouldAutoDismiss(false);
-    resetAutoDismiss();
-    resetErrorClearing();
-  }, [selectedHighlight]);
+    setImmediate(() => {
+      setIsFadingOut(false);
+      setShouldAutoDismiss(false);
+      resetErrorClearing();
+      resetAutoDismiss();
+    });
```

Comment:
I'm not happy with this solution, but im struggling to find an another way. Problem is that this runs before(?) `useOnDOMEvent`. The callback gets updated but `useOnDOMEvent` still has the stale value which causes it to dismiss it anyway.

Reply:
i'm not understanding the problem you're having. what callback gets updated? what updates it? what value is stale?

Reply:
I'm not exactly sure if i understand correctly whats happening.
The `useEffect` should basically reset the timeouts and state so that the component is renewed.
If the error is shown, and the click causes the same error:
1. First event listener for `useOnDOMEvent` executes which i expect
2. Then there is no rerender which im not sure why
3. Then the useEffect runs and resets the component
4. Then rerender happens which the correct values
5. Then an another rerender happens with values like the 1) was executed now which causes the fade out to start immediately.

Im experimenting now with the `useOnDOMEvent` to listen to mousedown rather than click.

Reply:
if you write a failing test i will help to figure this out, but if you want to try a few other things on your own, i would try:
- rearranging the order of the stuff in the useEffect, react usually batches state updates, but if it doesn't i could see that affecting things
- useEffect runs async while useLayoutEffect runs directly after the dom update, you might have better luck with useLayoutEffect in this case

i suspect that you're getting two renders because of the async nature of useEffect, you're getting one render as a result of the click event callback, and another render from the useEffect callback, and their state changes are clobbering each other. useState has a thing for when your new state is a derivative of your old state, you can pass a function into the update function so that your initial values are guaranteed to be consistent at the time that react decides to run the update, but this isn't helpful in your case because your new state is a derivative of _other_ states in your component, if you were to combine your states into one `useState` call, you might be able to use that feature and have more consistent behavior. the recommendation is in general to use separate `useState` calls, but because your states interact with one another, i think it is a special case.